### PR TITLE
DYN-2020 - Add test for package dependency state change.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR is to add a new test for the package dependency state change. To test this, I moved the new package to a different folder and loaded that package during the test. 

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang

### CODE REVIEW

File: test/DynamoCoreTests/PackageDependencyTests.cs [hunk 1]
```diff
@@ -457,17 +457,53 @@ public void PackageDependencyStateTest()
...
+            Assert.AreEqual(PackageDependencyState.Missing, firstPackage.State);
+
+            // Check for Warning package state, where the actually package is missing
+            // but the nodes are resolved by a different package.
+            PackageDependencyInfo secondPackage = (PackageDependencyInfo)packageDependenciesList[1];
+            Assert.AreEqual(new PackageDependencyInfo("Clockwork for Dynamo 2.x", new Version("2.1.2")), secondPackage);
+            Assert.AreEqual(PackageDependencyState.Warning, secondPackage.State);
+
+            // Check for Missing package state
+            PackageDependencyInfo thirdPackage = (PackageDependencyInfo)packageDependenciesList[2];
```
Comment (QilongTang) [inline test/DynamoCoreTests/PackageDependencyTests.cs]: @reddyashish Did you figure if you could call `LoadPackage` in the middle of the test for the missing case?

Comment (reddyashish) [reply test/DynamoCoreTests/PackageDependencyTests.cs]: The state doesn't change then because the file has to be reopened again to get the updated data.

Comment (reddyashish) [reply test/DynamoCoreTests/PackageDependencyTests.cs]: Updated the test to check for the state change in the same test. Dynamo restart is not needed.

**Comment by QilongTang** [comment]
LGTM after the modification, Thanks for updating it!

**Comment by QilongTang** [comment]
@reddyashish If your change is in the 2.4 branch, do you mind cherry-picking this as well?

**Comment by reddyashish** [comment]
Yes, I will cherry-pick it into 2.4.

**Pull request merged**